src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
> Would be interesting to find some documentation

[E0106](https://doc.rust-lang.org/error-index.html#E0106) description has formal rules of lifetime elision.

> We can also scale this PR to fields

I totally agree. I'll remove the return types analysis.